vars/provisionResources.groovy

```diff
@@ -67,19 +67,21 @@ def call(Map params, String region){
 
     if [[ -n "${params.scylla_ami_id ? params.scylla_ami_id : ''}" ]] ; then
         export SCT_AMI_ID_DB_SCYLLA="${params.scylla_ami_id}"
-    elif [[ -n "${params.gce_image_db ? params.gce_image_db : ''}" ]] ; then
+    fi
+    if [[ -n "${params.gce_image_db ? params.gce_image_db : ''}" ]] ; then
         export SCT_GCE_IMAGE_DB="${params.gce_image_db}"
-    elif [[ -n "${params.azure_image_db ? params.azure_image_db : ''}" ]] ; then
+    fi
+    if [[ -n "${params.azure_image_db ? params.azure_image_db : ''}" ]] ; then
         export SCT_AZURE_IMAGE_DB="${params.azure_image_db}"
-    elif [[ -n "${params.scylla_version ? params.scylla_version : ''}" ]] ; then
+    fi
+    if [[ -n "${params.scylla_version ? params.scylla_version : ''}" ]] ; then
         export SCT_SCYLLA_VERSION="${params.scylla_version}"
-    elif [[ -n "${params.scylla_repo ? params.scylla_repo : ''}" ]] ; then
+    fi
+    if [[ -n "${params.scylla_repo ? params.scylla_repo : ''}" ]] ; then
         export SCT_SCYLLA_REPO="${params.scylla_repo}"
-    elif [[ "${params.backend ? params.backend : ''}" == *"k8s"* ]] ; then
-        echo "Kubernetes backend can have empty scylla version. It will be taken from defaults of the scylla helm chart"
-    else
-        echo "need to choose one of SCT_AMI_ID_DB_SCYLLA | SCT_SCYLLA_VERSION | SCT_SCYLLA_REPO | SCT_GCE_IMAGE_DB"
```

Comment:
it's bit problematic we take those out, now if we have conflicting parameters like that, we might not notice it...

this kind of issue with the artifact tests, waste me lots of hours, until I've figure I've test the wrong thing (I've passing both scylla_repo and unified_packgae, and we don't have any code that warns you about it's conflicting)

Reply:
If you find a way to check env vars at this point then let me know.

Without ability to check env vars setting explicitly in pipeline we should either hack with writing to files (ugly, IMHO) or remove these exclusive checks (current variant).

Reply:
I'm not following exactly

I understand this current logic, doesn't apply for the case you are doing.

I don't understand why we can't change the logic to fit, and how reading environment variable is related to that (and what is the problem preventing reading environment variables)

I'm arguing that this check is important, and taking it out, would could lead to confusion if one didn't supplied information about which scylla he wants to run (i.e. the default when you trigger a job)

Reply:
> I'm not following exactly
> 
> I understand this current logic, doesn't apply for the case you are doing.

We get failures not having any of the required parameters. These changes are a must.

> I don't understand why we can't change the logic to fit, and how reading environment variable is related to that (and what is the problem preventing reading environment variables)

This is the jenkins bug that complicates things: https://issues.jenkins.io/browse/JENKINS-28921

> I'm arguing that this check is important, and taking it out, would could lead to confusion if one didn't supplied information about which scylla he wants to run (i.e. the default when you trigger a job from jenkins)

We have checks in SCT itself.

Reply:
well, we'll need to count on Argus to have those check as well.

and use argus as to avoid those silly mistakes